### ISSUE TITLE
Admin - Product Page - Message: "Upload JPEG files that are 1000 x 1000 pixels or larger (max. 2 GB). How to prepare images?" is repeating over and over again

### ISSUE THREAD

**Issue by LucianHus** [woothemes/woocommerce#34728]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

After updating to WooCommerce 6.9.2, in admin - product page - product image section - Message: "Upload JPEG files that are 1000 x 1000 pixels or larger (max. 2 GB). How to prepare images?" is repeating over and over again.
Prinnt screen: https://ibb.co/7SyDKgd

### Expected behavior

Message: "Upload JPEG files that are 1000 x 1000 pixels or larger (max. 2 GB). [How to prepare images?](https://woocommerce.com/posts/fast-high-quality-product-photos/)" should only be displayed once.

### Actual behavior

After updating to WooCommerce 6.9.2, in admin - product page - product image section - Message: "Upload JPEG files that are 1000 x 1000 pixels or larger (max. 2 GB). How to prepare images?" is repeating over and over again.

### Steps to reproduce

1. Go to admin
2. Product page
3. Image section, under Product image >> https://ibb.co/7SyDKgd

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): https://asterpremium.ro
Site address (URL): https://asterpremium.ro
WC Version: 6.9.2
REST API Version: ✔ 6.9.2
WC Blocks Version: ✔ 8.3.2
Action Scheduler Version: ✔ 3.4.0
Log Directory Writable: ✔
WP Version: 6.0.2
WP Multisite: –
WP Memory Limit: 4 GB
WP Debug Mode: –
WP Cron: ✔
Language: ro_RO
External object cache: –

### Server Environment ###

Server Info: LiteSpeed
PHP Version: 7.4.30
PHP Post Max Size: 128 MB
PHP Time Limit: 300
PHP Max Input Vars: 1000
cURL Version: 7.61.1
OpenSSL/1.1.1k

SUHOSIN Installed: –
MySQL Version: 8.0.30
Max Upload Size: 128 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 6.9.2
WC Database Prefix: wp_
Total Database Size: 79.98MB
Database Data Size: 60.66MB
Database Index Size: 19.32MB
wp_woocommerce_sessions: Data: 13.80MB + Index: 0.44MB + Engine MyISAM
wp_woocommerce_api_keys: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_attribute_taxonomies: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_downloadable_product_permissions: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_order_items: Data: 0.33MB + Index: 0.16MB + Engine MyISAM
wp_woocommerce_order_itemmeta: Data: 2.07MB + Index: 1.44MB + Engine MyISAM
wp_woocommerce_tax_rates: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_woocommerce_tax_rate_locations: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_shipping_zones: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_shipping_zone_locations: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_shipping_zone_methods: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_payment_tokens: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_payment_tokenmeta: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_woocommerce_log: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_actionscheduler_actions: Data: 0.15MB + Index: 0.09MB + Engine MyISAM
wp_actionscheduler_claims: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_actionscheduler_groups: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_actionscheduler_logs: Data: 0.12MB + Index: 0.11MB + Engine MyISAM
wp_adtribes_my_conversions: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_aioseo_cache: Data: 4.36MB + Index: 5.02MB + Engine InnoDB
wp_aioseo_links: Data: 0.08MB + Index: 0.03MB + Engine InnoDB
wp_aioseo_links_suggestions: Data: 0.52MB + Index: 0.05MB + Engine InnoDB
wp_aioseo_notifications: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_aioseo_posts: Data: 1.48MB + Index: 0.11MB + Engine InnoDB
wp_aioseo_redirects: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_aioseo_redirects_404: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_aioseo_redirects_404_logs: Data: 3.02MB + Index: 1.27MB + Engine InnoDB
wp_aioseo_redirects_hits: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_aioseo_redirects_logs: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_aioseo_terms: Data: 0.11MB + Index: 0.02MB + Engine InnoDB
wp_commentmeta: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_comments: Data: 0.73MB + Index: 0.25MB + Engine MyISAM
wp_gdpr_cc_options: Data: 0.01MB + Index: 0.00MB + Engine MyISAM
wp_hollerbox_stats: Data: 0.11MB + Index: 0.11MB + Engine InnoDB
wp_imagify_files: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_imagify_folders: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_links: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_litespeed_url: Data: 0.16MB + Index: 0.20MB + Engine InnoDB
wp_litespeed_url_file: Data: 0.06MB + Index: 0.14MB + Engine InnoDB
wp_loginizer_logs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailchimp_carts: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_mailchimp_jobs: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_ms_snippets: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_options: Data: 13.51MB + Index: 0.34MB + Engine MyISAM
wp_postmeta: Data: 11.66MB + Index: 4.57MB + Engine MyISAM
wp_posts: Data: 2.53MB + Index: 0.47MB + Engine MyISAM
wp_snippets: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_social_users: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_term_relationships: Data: 0.16MB + Index: 0.34MB + Engine MyISAM
wp_term_taxonomy: Data: 0.04MB + Index: 0.06MB + Engine MyISAM
wp_termmeta: Data: 0.11MB + Index: 0.09MB + Engine MyISAM
wp_terms: Data: 0.03MB + Index: 0.08MB + Engine MyISAM
wp_trp_gettext_ro_ro: Data: 0.63MB + Index: 0.88MB + Engine MyISAM
wp_trp_original_meta: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_trp_original_strings: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_usermeta: Data: 1.09MB + Index: 0.76MB + Engine MyISAM
wp_users: Data: 0.07MB + Index: 0.08MB + Engine MyISAM
wp_vtprd_purchase_log: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_vtprd_purchase_log_product: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_vtprd_purchase_log_product_rule: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_vtprd_transient_cart_data: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.01MB + Engine MyISAM
wp_wc_admin_notes: Data: 0.04MB + Index: 0.00MB + Engine MyISAM
wp_wc_category_lookup: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_wc_customer_lookup: Data: 0.09MB + Index: 0.09MB + Engine MyISAM
wp_wc_download_log: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wc_euplatesc_trid: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wc_order_coupon_lookup: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wc_order_product_lookup: Data: 0.31MB + Index: 0.26MB + Engine MyISAM
wp_wc_order_stats: Data: 0.10MB + Index: 0.07MB + Engine MyISAM
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine MyISAM
wp_wc_product_attributes_lookup: Data: 0.11MB + Index: 0.19MB + Engine MyISAM
wp_wc_product_download_directories: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.07MB + Index: 0.08MB + Engine MyISAM
wp_wc_rate_limits: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wc_reserved_stock: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wc_tax_rate_classes: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_wc_webhooks: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wdp_order_items: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wdp_orders: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wdp_rules: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wdr_order_discounts: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wdr_order_item_discounts: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wdr_rules: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpmailsmtp_debug_events: Data: 0.22MB + Index: 0.00MB + Engine InnoDB
wp_wpmailsmtp_tasks_meta: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpr_rucss_resources: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpr_rucss_used_css: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_yith_wcwl: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_yith_wcwl_lists: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_yoast_indexable: Data: 2.14MB + Index: 0.40MB + Engine MyISAM
wp_yoast_indexable_hierarchy: Data: 0.07MB + Index: 0.17MB + Engine MyISAM
wp_yoast_migrations: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_yoast_primary_term: Data: 0.04MB + Index: 0.04MB + Engine MyISAM
wp_yoast_prominent_words: Data: 0.29MB + Index: 0.51MB + Engine MyISAM
wp_yoast_seo_links: Data: 0.06MB + Index: 0.02MB + Engine MyISAM

### Post Type Counts ###

attachment: 1590
blocks: 10
custom_css: 1
customize_changeset: 22
hollerbox: 3
language_switcher: 2
nav_menu_item: 40
oembed_cache: 6
page: 18
popup: 2
popup_theme: 8
post: 18
product: 723
product_variation: 18
revision: 219
seedprod: 1
shop_coupon: 3
shop_order: 1482
ux_template: 14
wc_dynamic_pricing: 1
wp_global_styles: 1
wpcf7_contact_form: 6

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (38) ###

ManageWP - Worker: by GoDaddy – 4.9.14
Age Gate: by Phil Baker – 3.0.4
AIOSEO - Image SEO: by All in One SEO Team – 1.1.0.1
AIOSEO - IndexNow: by All in One SEO Team – 1.0.5
AIOSEO - Link Assistant: by All in One SEO Team – 1.0.9
AIOSEO - Local Business: by All in One SEO Team – 1.2.10
AIOSEO - News Sitemap: by All in One SEO Team – 1.0.9
AIOSEO - Redirects: by All in One SEO Team – 1.2.3
AIOSEO - REST API: by All in One SEO Team – 1.0.2
AIOSEO - Video Sitemap: by All in One SEO Team – 1.1.6
Akismet Anti-Spam: by Automattic – 5.0
All in One SEO Pro: by All in One SEO Team – 4.2.5.1
Code Snippets: by Code Snippets Pro – 3.2.0
Contact Form 7 Extension For Mailchimp: by Renzo Johnson – 0.5.62
Contact Form 7: by Takayuki Miyoshi – 5.6.3
GDPR Cookie Compliance: by Moove Agency – 4.8.12
HollerBox: by Groundhogg Inc. – 2.0.14
Imagify: by Imagify - Optimize Images & Convert WebP – 2.0
Import and export users and customers: by codection – 1.20.4
iThemeland Free Gifts For WooCommerce: by iThemelandco – 1.7.5
Jetpack: by Automattic – 11.3.1
LiteSpeed Cache: by LiteSpeed Technologies – 5.2.1
Loginizer: by Softaculous – 1.7.4
Nextend Social Login: by Nextendweb – 3.1.6
Safe SVG: by 10up – 2.0.3
Send email to customer on cancelled order in WooCommerce: by Laura Díaz – 1.1
Advanced Editor Tools (previously TinyMCE Advanced): by Automattic – 5.6.0
TranslatePress - SEO Pack Add-on: by Cozmoslabs
Razvan Mocanu – 1.4.5

TranslatePress - Multilingual: by Cozmoslabs
Razvan Mocanu
Madalin Ungureanu
Cristophor Hurduban – 2.3.7

WooCommerce Euplatesc Gateway: by Zamfir Vladimir – 2.3.8
Checkout Field Editor for WooCommerce: by ThemeHigh – 1.7.2
Woo Product Category Discount: by WooExtend – 4.3
Product Feed PRO for WooCommerce: by AdTribes.io – 11.8.7
Pixel Manager for WooCommerce: by SweetCode – 1.21.0
WooCommerce: by Automattic – 6.9.2
WP Mail SMTP: by WPForms – 3.5.2
ReCaptcha v2 for Contact Form 7: by IQComputing – 1.4.3
YITH WooCommerce Wishlist: by YITH – 3.12.0

### Inactive Plugins (1) ###

WP Rocket: by WP Media – 3.11.2

### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
maintenance.php: maintenance.php

### Must Use Plugins (1) ###

ManageWP - Worker Loader: by GoDaddy –

### Settings ###

API Enabled: –
Force SSL: –
Currency: RON (lei)
Currency Position: right_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: –

### WC Pages ###

Shop base: #15 - /magazin-online/
Cart: #16 - /cos-de-cumparaturi/
Checkout: #17 - /detalii-comanda/
My account: #18 - /cont-client/
Terms and conditions: #549 - /termeni-si-conditii/

### Theme ###

Name: Flatsome Child
Version: 3.0
Author URL: 
Child Theme: ✔
Parent Theme Name: Flatsome
Parent Theme Version: 3.16.0
Parent Theme Author URL: https://uxthemes.com
WooCommerce Support: ✔

### Templates ###

Overrides: flatsome/woocommerce/archive-product.php
flatsome/woocommerce/cart/cart-empty.php
flatsome/woocommerce/cart/cart-shipping.php
flatsome/woocommerce/cart/cart.php
flatsome/woocommerce/checkout/form-checkout.php
flatsome/woocommerce/checkout/form-coupon.php
flatsome/woocommerce/checkout/terms.php
flatsome/woocommerce/checkout/thankyou.php
flatsome/woocommerce/content-product-cat.php
flatsome/woocommerce/content-product.php
flatsome/woocommerce/content-single-product.php
flatsome/woocommerce/content-widget-product.php
flatsome/woocommerce/global/breadcrumb.php
flatsome/woocommerce/global/quantity-input.php
flatsome/woocommerce/global/wrapper-end.php
flatsome/woocommerce/global/wrapper-start.php
flatsome/woocommerce/loop/loop-end.php
flatsome/woocommerce/loop/loop-start.php
flatsome/woocommerce/loop/pagination.php
flatsome/woocommerce/loop/rating.php
flatsome/woocommerce/loop/result-count.php
flatsome/woocommerce/loop/sale-flash.php
flatsome/woocommerce/myaccount/form-login.php
flatsome/woocommerce/myaccount/navigation.php
flatsome/woocommerce/notices/error.php
flatsome/woocommerce/notices/notice.php
flatsome/woocommerce/notices/success.php
flatsome/woocommerce/product-searchform.php
flatsome/woocommerce/single-product/photoswipe.php
flatsome/woocommerce/single-product/price.php
flatsome/woocommerce/single-product/product-image.php
flatsome/woocommerce/single-product/product-thumbnails.php
flatsome/woocommerce/single-product/rating.php
flatsome/woocommerce/single-product/related.php
flatsome/woocommerce/single-product/review.php
flatsome/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php
flatsome/woocommerce/single-product/short-description.php
flatsome/woocommerce/single-product/tabs/tabs.php
flatsome/woocommerce/single-product/title.php
flatsome/woocommerce/single-product/up-sells.php
flatsome/woocommerce/single-product-reviews.php
flatsome/woocommerce/single-product.php
flatsome/woocommerce/content-product_cat.php

### Admin ###

Enabled Features: activity-panels
analytics
coupons
customer-effort-score-tracks
experimental-products-task
experimental-import-products-task
experimental-fashion-sample-products
experimental-product-tour
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: minified-js
new-product-management-experience
settings

Daily Cron: ✔ Next scheduled: 2022-09-19 15:01:55 +00:00
Options: ✔
Notes: 104
Onboarding: completed

### Action Scheduler ###

Complete: 494
Oldest: 2022-09-13 07:19:29 +0000
Newest: 2022-09-19 10:06:25 +0000

Pending: 8
Oldest: 2022-09-19 10:16:26 +0000
Newest: 2022-09-23 17:01:32 +0000

### Status report information ###

Generated at: 2022-09-19 13:10:25 +03:00
`

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by mattsherman** [comment]
@LucianHus Thanks for reporting this issue. I have verified that it is reproducible in certain scenarios. Most likely, it is triggered by a plugin.

We will fix this issue in a future version of WooCommerce.

**Comment by mattsherman** [comment]
Additional reports of this by users can be found in the comments of #33660

**Comment by mattsherman** [comment]
@LucianHus If you have a test/staging site, could you please disable all plugins other than WooCommerce and confirm if the issue still exists?

If it does not, please re-enable plugins one by one, checking after each one is re-enabled whether the issue exists.

This will help us confirm with even greater certainty that the fix we have in #34739 addresses your issue.

Thanks!

**Comment by LucianHus** [comment]
@mattsherman - I have found the cause: disabling All in One SEO Pro plugin will eliminate this error (at least in case of the shops managed by me).

**Comment by arnaudbroes** [comment]
Hey @mattsherman

AIOSEO dev here. I happened to be working on this issue just an hour ago.

I found that this is happening because WooCommerce has added this filter in the constructor of the `WP_Product_Simple` class.
`add_filter('admin_post_thumbnail_html', array( $this, 'add_product_photo_suggestions' ) );`

Because we call the `wc_get_product()` method a bunch of times when the post editor is loading, WooCommerce is enqueuing an alert with each call because it instantiates a new instance of the class each time. I've worked around this by creating a wrapper method in our own plugin that subsequently removes the filter every time we call `wc_get_product()`.

```
$product = wc_get_product( $postId );

remove_filter( 'admin_post_thumbnail_html', [ $product, 'add_product_photo_suggestions' ] );

return $product;
```

What really ought to happen is that WooCommerce improves this on their end by ensuring that their callback only runs once since I'm betting there are many more plugins out there that use `wc_get_product()` in the admin/post editor. We just happen to be a prominent example. Ideally `wc_get_product()` would actually just have a cache for the products that are being returned (because why instantiate identical objects multiple times?) but I assume that's a bigger rewrite.

(@LucianHus we'll fix this in version 4.2.6 which will be released early October, but feel free to [reach out to our support team](https://aioseo.com/contact) in case you'd like to receive a beta build with this fix).

**Comment by mattsherman** [comment]
@LucianHus Thanks for narrowing that down!

@arnaudbroes 

> What really ought to happen is that WooCommerce improves this on their end by ensuring that their callback only runs once since I'm betting there are many more plugins out there that use `wc_get_product()` in the admin/post editor. We just happen to be a prominent example. 

I totally agree. My pending fix in #34739 will not require plugin devs such as yourself to implement any workarounds.

**Comment by mattsherman** [comment]
Additional reports from users in the forums:

https://wordpress.org/support/topic/product-image-bug/
https://wordpress.org/support/topic/request-to-remove-hints-on-product-images-too-long/

Also, note that this issue was originally reported back in August, with a slightly different scenario:

https://github.com/woocommerce/woocommerce/issues/34256

**Comment by mattsherman** [comment]
This has been fixed in [WooCommerce 6.9.3](https://developer.woocommerce.com/2022/09/20/woocommerce-6-9-3-fix-release/)
